includes/specials/SpecialUndelete.php

```diff
@@ -328,9 +328,9 @@ function isDeleted() {
 	function undelete( $timestamps, $comment = '', $fileVersions = array(), $unsuppress = false ) {
 		global $wgUser;
 
-		$resultMock = '';
-		if ( !$this->title->userCan( 'undelete' ) ) {
```

Comment:
getPermissionsErrors allows us to pass errors returned from hooks to user, while userCan calls same logic but maps result to boolean saying is there where any errors.
So in general if we wany to just check permissions we chould use userCan but if we want to throw it we should check using getPermissionsErrors